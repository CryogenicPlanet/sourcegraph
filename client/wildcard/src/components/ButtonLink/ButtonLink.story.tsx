import { Meta } from '@storybook/react'
import React from 'react'

import { BrandedStory } from '@sourcegraph/branded/src/components/BrandedStory'
import webStyles from '@sourcegraph/web/src/SourcegraphWebApp.scss'

import { ButtonLink } from '.'

const Story: Meta = {
    title: 'wildcard/ButtonLink',

    decorators: [
        story => (
            <BrandedStory styles={webStyles}>{() => <div className="container mt-3">{story()}</div>}</BrandedStory>
        ),
    ],

    parameters: {
        component: ButtonLink,
    },
}

export default Story

export const Simple = () => (
    <>
        <h3>Simple button link</h3>
        <ButtonLink to="#">Click me!</ButtonLink>

        <h3>Disabled button link</h3>
        <ButtonLink className="btn" disabled={true}>
            Disabled button
        </ButtonLink>
        <h3>Button link with undefined to property</h3>
        <ButtonLink>Click me</ButtonLink>
        <h3>Button link with onSelect</h3>
        <ButtonLink onSelect={() => alert('Link button selected')}>Click me</ButtonLink>
    </>
)
