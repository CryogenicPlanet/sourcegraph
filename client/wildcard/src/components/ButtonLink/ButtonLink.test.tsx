import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import sinon from 'sinon'

import { ButtonLink } from './ButtonLink'

describe('Button link', () => {
    test('renders correctly', () => {
        const { asFragment } = render(<ButtonLink to="http://example.com">Button link</ButtonLink>)
        expect(asFragment()).toMatchSnapshot()
    })
    test('Works when to is undefined', () => {
        const { container } = render(<ButtonLink>I am a link</ButtonLink>)
        expect(container.firstChild).toMatchInlineSnapshot(`
            <a
              class="anchorLink nav-link"
              href=""
              role="button"
              tabindex="0"
            >
              I am a link
            </a>
        `)
    })
    test('Works with onSelect', () => {
        const handleClick = sinon.spy()
        render(<ButtonLink onSelect={handleClick}>Click Me</ButtonLink>)
        fireEvent.click(screen.getByText(/click me/i))
        sinon.assert.called(handleClick)
    })
    test('Should be disabled with disabled prop', () => {
        render(
            <ButtonLink className="btn" disabled={true}>
                Disabled button
            </ButtonLink>
        )
        expect(screen.getByText(/disabled button/i)).toHaveClass('disabled')
    })
})
