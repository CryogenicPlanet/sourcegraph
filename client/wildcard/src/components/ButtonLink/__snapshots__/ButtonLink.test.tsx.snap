// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button link renders correctly 1`] = `
<DocumentFragment>
  <a
    class="anchorLink nav-link"
    href="http://example.com"
    tabindex="0"
  >
    Button link
  </a>
</DocumentFragment>
`;
